sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -755,12 +751,11 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 			sof_rt1015p_dai_link(&links[id]);
 		} else if (sof_rt5682_quirk & SOF_RT1019_SPEAKER_AMP_PRESENT) {
 			sof_rt1019p_dai_link(&links[id]);
-		} else if (sof_rt5682_quirk &
-				SOF_MAX98373_SPEAKER_AMP_PRESENT) {
-			links[id].codecs = max_98373_components;
-			links[id].num_codecs = ARRAY_SIZE(max_98373_components);
-			links[id].init = max_98373_spk_codec_init;
-			links[id].ops = &max_98373_ops;
+		} else if ((sof_rt5682_quirk &
+				SOF_MAX98373_SPEAKER_AMP_PRESENT) ||
+			(sof_rt5682_quirk &
+				SOF_MAX98390_SPEAKER_AMP_PRESENT)) {
```

Comment:
Can we merge `SOF_MAX98373_SPEAKER_AMP_PRESENT` and `SOF_MAX98390_SPEAKER_AMP_PRESENT`? Like #define SOF_MAXIM_SPEAKER_AMP_PRESENT	BIT(29)

Reply:
considering unique codes for max98373 here, so separating them out still. @bardliao any comments?

        if (sof_rt5682_quirk & SOF_MAX98373_SPEAKER_AMP_PRESENT) {
                /* Disable Left and Right Spk pin after boot */
                snd_soc_dapm_disable_pin(dapm, "Left Spk");
                snd_soc_dapm_disable_pin(dapm, "Right Spk");
                err = snd_soc_dapm_sync(dapm);
                if (err < 0)
                        return err;
        }